### PR TITLE
chore(infra): Deploy domain app on a separate instance and enable background jobs on it

### LINKED ISSUE #3801: Move background jobs to a separate node
We don't want sync and other background jobs to overload the repo pool or other VM resources in web/api nodes, so they should be moved to a separate app and deployed to separate VMs.

### PR DESCRIPTION
**Pull request by AndrewDryga**
Closes #3801

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -107,6 +107,180 @@ services:
...
+      - app
+
+  domain:
+    build:
+      context: elixir
+      cache_from:
+        - type=registry,ref=us-east1-docker.pkg.dev/firezone-staging/cache/domain:main
+      args:
+        APPLICATION_NAME: domain
+    image: ${API_IMAGE:-us-east1-docker.pkg.dev/firezone-staging/cache/domain}:${API_TAG:-main}
```
Comment (jamilbk) [inline docker-compose.yml]:
```suggestion
    image: ${API_IMAGE:-us-east1-docker.pkg.dev/firezone-staging/firezone/domain}:${API_TAG:-main}
```

File: docker-compose.yml [hunk 2]
```diff
@@ -107,6 +107,180 @@ services:
...
       - app
 
+  api:
+    build:
+      context: elixir
+      cache_from:
+        - type=registry,ref=us-east1-docker.pkg.dev/firezone-staging/cache/api:main
+      args:
+        APPLICATION_NAME: api
+    image: ${API_IMAGE:-us-east1-docker.pkg.dev/firezone-staging/cache/api}:${API_TAG:-main}
```
Comment (jamilbk) [inline docker-compose.yml]:
```suggestion
    image: ${API_IMAGE:-us-east1-docker.pkg.dev/firezone-staging/firezone/api}:${API_TAG:-main}
```

I think I typo'd these in a previous PR

**Review by jamilbk — CHANGES_REQUESTED** [review]
You'll also need to update `.github/workflows/_integration_tests.yml` to allow overriding the domain_image and tag like the others.

**Comment by jamilbk** [comment]
The perf test image build will fail because we're changing CI config but the perf tests are not required for merge so we should be ok.

**Review by jamilbk — COMMENTED** [review]
Sorry just remembered that we may want to start domain for perf tests too?
https://github.com/firezone/firezone/blob/main/.github/workflows/ci.yml#L188

**Comment by AndrewDryga** [comment]
@jamilbk it's done, `perf` will fail because there is no such container on the main.

**Pull request merged**